edgeless_orc/src/orchestration_logic.rs

```diff
@@ -0,0 +1,53 @@
+use rand::{rngs::StdRng, seq::SliceRandom, SeedableRng};
+use uuid::Uuid;
+
+use crate::OrchestrationStrategy;
+
+/// Keeps all the necessary state that is needed to make simple orchestration
+/// decisions. Provides convenience methods that can be used by the
+/// orchestrator.
+pub struct OrchestrationLogic {
+    orchestration_strategy: OrchestrationStrategy,
+    round_robin_current_index: usize,
+    rng: StdRng,
+    orchestrated_nodes: Vec<Uuid>,
+}
+
+impl OrchestrationLogic {
+    pub fn new(orchestration_strategy: Option<OrchestrationStrategy>, orchestrated_nodes: Vec<Uuid>) -> Self {
+        Self {
+            orchestration_strategy: match orchestration_strategy {
+                Some(s) => s,
+                None => OrchestrationStrategy::Random,
+            },
+            round_robin_current_index: 0,
+            rng: StdRng::from_entropy(),
+            orchestrated_nodes,
+        }
+    }
+}
+
+/// This iterator can be used to select the next node on which a function
+/// instance should be spawned, based on a general orchestration strategy as
+/// defined in the settings.
+impl Iterator for OrchestrationLogic {
+    type Item = Uuid;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match self.orchestration_strategy {
+            OrchestrationStrategy::Random => {
```

Comment:
guess this is a first start, but I would implement that using a trait that is implemented by a strategy.
Also there might be pinning from the annotations.

Reply:
Yes, totally. This is just the first thing that came to my mind. What do you mean with "pinning from the annotations"?

Reply:
There should be an annotation to pin a function instance to a fixed node.

Reply:
Yes, correct @raphaelhetzel . Hope the discussion on the list of supported annotations will take off soon enough.